### PR TITLE
Open Minicart when click on the Buy Button

### PR DESCRIPTION
**Pull request by emersonlaurentino**
## What's the purpose of this pull request?
We transferred the minicart status control to a provider and added a new attribute in the `userBuyButton` hook to open the minicart when true.

## How it works? 
Context

## How to test it?
https://github.com/vtex-sites/marinbrasil.store/pull/212

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

File: packages/gatsby-theme-instore/src/typings/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+declare type Maybe<T> = T | undefined | null
```
Comment (lucis) [inline packages/gatsby-theme-instore/src/typings/index.d.ts:1]: Why this file?

**Review by lucis — APPROVED** [review]

File: packages/gatsby-theme-instore/src/typings/index.d.ts (hunk 1 above)
Comment (emersonlaurentino) [reply packages/gatsby-theme-instore/src/typings/index.d.ts:1]: I don't know, this file came when I run `yarn upgrade`

File: packages/gatsby-theme-vtex/src/sdk/minicart/index.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+export const MinicartState = createContext<MinicartStateContext>({
+  isOpen: false,
+  toggle: () => null,
+})
+
+export const MinicartProvider: FC = ({ children }) => {
+  const [isOpen, setOpen] = useState<boolean>(false)
+  const toggle = () => setOpen(!isOpen)
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/minicart/index.tsx:15]: please useCallback in here

File: packages/gatsby-theme-vtex/src/components/Minicart/index.tsx [hunk 3]
```diff
@@ -22,11 +22,11 @@ const Minicart: FC = () => {
   return (
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/components/Minicart/index.tsx:22]: can you also movie this customVariant variable outside this component ?

**Review by tlgimenes — APPROVED** [review]
LGTM after the requested changes

**Pull request merged**